make-rules/ips.mk

```diff
@@ -194,12 +194,15 @@ PUBLISH_STAMP ?= $(BUILD_DIR)/.published-$(MACH)
 
 # Do all that is needed to ensure the package is consistent for publishing,
 # except actually pushing to a repo, separately from the push to the repo.
-pre-publish:	build install $(PRE_PUBLISH_STAMP)
+# Note that running "install" rule as a dependency for multiple targets can
+# have adverse effects, so instead we call it at most once by requiring the
+# touched ALL_INSTALLED_STAMP file.
+pre-publish:	$(ALL_INSTALLED_STAMP) $(PRE_PUBLISH_STAMP)
```

Comment:
I thought about it for a while. Logically, this should not be a problem, if for packaging we depend on stabilizing actually the "install" results. That should call "build", but as an internal implementation detail, IMHO.

But according to https://github.com/OpenIndiana/oi-userland/commit/4221e17e0bba8bf0ee49ce857a6d61200767cbf7 the older state (before introducing pre-publish rule), the "build" was explicitly required: "-publish:		build install $(PUBLISH_STAMP)"

So I guess I'll add the "build" back to dependencies of ALL_INSTALLED_STAMP where it was initially. At least for now it makes sense, since the main use-case is same ips.mk.